firebase-firestore/src/main/java/com/google/firebase/firestore/util/ListenerRegistrationImpl.java

```diff
@@ -33,6 +58,119 @@
+    }
+
+    final CallbackList callbacks = new CallbackList();
+  }
+
+  @SuppressWarnings("deprecation")
+  public static class StopListenerFragment extends android.app.Fragment {
+    @Override
+    public void onStop() {
+      super.onStop();
+      callbacks.run();
+    }
+
+    final CallbackList callbacks = new CallbackList();
+  }
+
+  @Nullable
+  private static <T> T castFragment(Class<T> fragmentClass, @Nullable Object fragment, String tag) {
+    try {
+      return fragmentClass.cast(fragment);
+    } catch (ClassCastException e) {
+      throw new IllegalStateException(
+          "Fragment with tag '"
+              + tag
+              + "' is a "
+              + fragment.getClass().getName()
+              + " but should be a "
+              + fragmentClass.getName());
+    }
+  }
+
+  private static final String SUPPORT_FRAGMENT_TAG = "FirestoreOnStopObserverSupportFragment";
+  private static final String FRAGMENT_TAG = "FirestoreOnStopObserverFragment";
+
+  /**
+   * Implementation for non-FragmentActivity Activity's. Unfortunatly, all Fragment related
+   * classes/methods with nonFragmentActivityActivity's are deprecated, implying that almost
+   * everything in this function is deprecated.
+   */
+  @SuppressWarnings("deprecation")
+  private void onActivityStopCallOnce(Activity activity, Runnable callback) {
+    hardAssert(
+        !(activity instanceof FragmentActivity),
+        "onActivityStopCallOnce must be called with a *non*-FragmentActivity Activity.");
+
+    activity.runOnUiThread(
+        () -> {
+          StopListenerFragment fragment =
+              castFragment(
+                  StopListenerFragment.class,
+                  activity.getFragmentManager().findFragmentByTag(FRAGMENT_TAG),
+                  FRAGMENT_TAG);
+
+          if (fragment == null || fragment.isRemoving()) {
+            fragment = new StopListenerFragment();
+            activity
+                .getFragmentManager()
+                .beginTransaction()
+                .add(fragment, FRAGMENT_TAG)
+                .commitAllowingStateLoss();
+
+            activity.getFragmentManager().executePendingTransactions();
+          }
+
+          fragment.callbacks.add(callback);
+        });
+  }
+
+  private void onFragmentActivityStopCallOnce(FragmentActivity activity, Runnable callback) {
+    activity.runOnUiThread(
+        () -> {
+          StopListenerSupportFragment fragment =
+              castFragment(
+                  StopListenerSupportFragment.class,
+                  activity.getSupportFragmentManager().findFragmentByTag(SUPPORT_FRAGMENT_TAG),
+                  SUPPORT_FRAGMENT_TAG);
+
+          if (fragment == null || fragment.isRemoving()) {
+            fragment = new StopListenerSupportFragment();
+            activity
+                .getSupportFragmentManager()
+                .beginTransaction()
+                .add(fragment, SUPPORT_FRAGMENT_TAG)
+                .commitAllowingStateLoss();
```

Comment:
You can use `commitNow()` on the **support** FragmentTransaction, but if you want to use this for consistency with the built-in FragmentTransaction that's fine.

Reply:
I think I'll stick with commtLater for consistency... especially since it's *so close* to the other, that the difference could easily be overlooked. (If we eventually decide to have different implementations, then leap-frogging direct to using android lifecycles seems to make sense.)